src/main/java/com/example/country/CountryService.java

```diff
@@ -40,4 +39,14 @@ public List<Country> getCountries(String countryStartsWith, String cityStartsWit
         }
     }
 
+    public Country insert(int countryCode, String country, String city) {
+        Country country1 = new Country(countryCode, country, city);
+        if (countryMapper.findByCountryCode(countryCode).isPresent()) {
+            throw new CountryDuplicatedException("Country with code " + countryCode + " duplicated");
+        }
+        countryMapper.insert(country1);
```

Comment:
先に存在チェックしてインスタンス化するほうが自然ですね。
細かいですがインスタンス化するのはメモリを消費するのでより効率を考えるという意味でもこの順番がいいです。
```suggestion
        if (countryMapper.findByCountryCode(countryCode).isPresent()) {
            throw new CountryDuplicatedException("Country with code " + countryCode + " duplicated");
        }
        Country country1 = new Country(countryCode, country, city);
        countryMapper.insert(country1);
```

Reply:
ご教示いただきありがとうございます。
修正しました。